src/main/java/de/ait/secondlife/services/ImageServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+package de.ait.secondlife.services;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import de.ait.secondlife.constants.EntityType;
+import de.ait.secondlife.constants.ImageConstants;
+import de.ait.secondlife.domain.dto.ImageCreateDto;
+import de.ait.secondlife.domain.dto.ImagePathsResponseDto;
+import de.ait.secondlife.domain.entity.EntityImage;
+import de.ait.secondlife.exception_handling.exceptions.bad_request_exception.BadFileFormatException;
+import de.ait.secondlife.exception_handling.exceptions.bad_request_exception.BadFileSizeException;
+import de.ait.secondlife.exception_handling.exceptions.bad_request_exception.FileNameIsWrongException;
+import de.ait.secondlife.exception_handling.exceptions.bad_request_exception.MaxImageCountException;
+import de.ait.secondlife.exception_handling.exceptions.bad_request_exception.is_null_exceptions.ParameterIsNullException;
+import de.ait.secondlife.exception_handling.exceptions.not_found_exception.FileNotFoundExecption;
+import de.ait.secondlife.exception_handling.exceptions.not_found_exception.ImagesNotFoundException;
+import de.ait.secondlife.repositories.ImageRepository;
+import de.ait.secondlife.services.interfaces.ImageService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.*;
+
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class ImageServiceImpl implements ImageService, ImageConstants {
+
+    private final AmazonS3 s3Client;
+    private final ImageRepository repository;
+
+    @Value("${BUCKET_NAME}")
+    private String bucketName;
+
+    @Value("${DIR_PREFIX}")
+    private String dirPrefix;
+
+    private final int MAX_IMAGE_COUNT = 5;
+
+    @Override
+    public void saveNewImage(ImageCreateDto dto) {
+        MultipartFile file = dto.getFile();
+        String entityType = EntityType.get(dto.getEntityCode()).getType();
+        Long entityId = dto.getEntityId();
+
+        ImagePathsResponseDto currentImages = findAllImageForEntity(entityType, entityId);
+        for (Set<String> set : currentImages.getImages().values()) {
+            if (set.size() >= MAX_IMAGE_COUNT) throw new MaxImageCountException(MAX_IMAGE_COUNT);
```

Comment:
here it is not very appropriate to have one max value for all types of entities. otherwise, the check does not work properly

Reply:
fix